src/models/user.js

```diff
@@ -1,15 +1,55 @@
 import crypto from 'crypto'
 import { Model } from 'sequelize'
+import jwtBuilder from '../JWTBuilder'
+import UsernameNotFoundError from '../errors/UsernameNotFoundError'
+import InvalidPasswordError from '../errors/InvalidPasswordError'
 
-export default (sequelize, DataTypes) => {
+export function buildUserModel (sequelizeModels) {
   class User extends Model {
-    async passwordMatches (value, models = sequelize.models) {
+    static async signin (username, password) {
+      const user = await User.findByUsername(username)
+
+      await user.validatePassword(password)
+
+      const jwt = await user.generateJWT()
+
+      return { user, jwt }
+    }
+
+    static async findByUsername (username) {
+      const user = await User.findOne({ where: { username } })
+
+      if (!user) {
+        throw new UsernameNotFoundError('Could not find a User with username: ' + username)
+      }
+
+      return user
+    }
+
+    async validatePassword (password) {
+      const passwordIsValid = await this.passwordMatches(password)
+
+      if (!passwordIsValid) {
+        throw new InvalidPasswordError('Invalid password')
```

Comment:
Here, since you're throwing an "InvalidPasswordError", the message 'invalid password' is redundant. You can have like a default message or even no message at all.